### PR TITLE
The Objectives of the Spider Clan

### PR DESCRIPTION
**Pull request by Kurfursten**
**INTENDED FOR MERGE AFTER #22321**

The Spider Clan wishes to dominate all technology. They will not tolerate Nanotrasen research into advanced technologies.

**If there is at least one active cyborg**, they will receive Dominate Silicons. If not, 60% chance to receive Delayed Assassinate, 40% chance to receive Skull Trophies
* Kill all Cyborgs OR
* Steal 2-4 (capped at number of cyborgs at time of issuance) posibrains/MMIs that contained players

**If there is at least one active AI**, they will receive Dominate AI, if not, they will receive Steal.
* Kill all AIs OR
* Steal a functional AI

**If there are 15 or less people**, 10% chance of Silence, otherwise, Survive.

Also includes code for Destroy Servers, but I don't think it's quite ready since Ninjas lack a good tool for destroying servers. Glove hacking? I'm not sure yet.

Additionally, ninjas get the new Cyber Trophy Belt instead of the Trophy Belt.

🆑 
* tweak: The Spider Clan has new objectives.
* rscadd: Ninjas are now equipped with Cyber Trophy Belts. They contain posibrains, MMIs, and AI cards, and force those within to speak to only the wearer.

### CODE REVIEW

File: code/datums/gamemode/role/ninja.dm [hunk 1]
```diff
@@ -18,10 +18,39 @@
...
+			AppendObjective(/datum/objective/target/delayed/assassinate)
+		else
+			AppendObjective(/datum/objective/target/skulls)
+
+	if(ai_list.len)
+		AppendObjective(/datum/objective/killorstealAI)
+	else
+		AppendObjective(/datum/objective/target/steal)
+
+	if(player_list.len<=15 && prob(10))
```
Comment (Kurfursten) [inline code/datums/gamemode/role/ninja.dm]: In retrospect this includes observers, dead, etc. so probably a better way is to pull the living playercount from the gamemode

Comment (SirSkeleto) [reply code/datums/gamemode/role/ninja.dm]: iterate & increment on some conditional?

**Comment by agentofzog2** [comment]
"Steal 2-4 (capped at number of cyborgs at time of issuance) posibrains/MMIs that contained players"

Does ninja even have a way to get those out without stealing a science ID and blowing/disassembling them?
Also in general I think giving a human antag an objective to kill something that can either be defeated with mere words if on asimov or is nearly impossible to defeat with the ninja's current gear if on anything else is a bad idea. Not to mention the metagamey awkwardness of 
Naruto says: Domo arigato mr roboto please let me into your core and turn off all turrets. 

Oh and does the cyber-trophy belt prevent the mmis from being tracked on the CMC?

**Comment by Kurfursten** [comment]
I'm going to PR to make borgs mashable more easily.

I'm not too worried about Asimov being inconvenient. Crew should choose lawset wisely if ninjas are about.

**Comment by Kurfursten** [comment]
* All the mobs inside the belt can now chat communally
* Belt now blocks CMC tracking
* Silence chance upped to 25% and only fires if there are no more than 15 living silicons and carbons on the station Z level with active clients.

**Pull request merged**